public/app/plugins/datasource/cloudwatch/components/MetricStatEditor/MetricStatEditor.test.tsx

```diff
@@ -40,18 +40,21 @@ describe('MetricStatEditor', () => {
     config.featureToggles.cloudWatchCrossAccountQuerying = originalFeatureToggleValue;
   });
   describe('statistics field', () => {
-    test.each(['Average', 'p23.23', 'p34', '$statistic'])('should accept valid values', async (statistic) => {
-      const onChange = jest.fn();
-
-      render(<MetricStatEditor {...props} onChange={onChange} />);
-
-      const statisticElement = await screen.findByLabelText('Statistic');
-      expect(statisticElement).toBeInTheDocument();
-
-      await userEvent.type(statisticElement, statistic);
-      fireEvent.keyDown(statisticElement, { keyCode: 13 });
-      expect(onChange).toHaveBeenCalledWith({ ...props.metricStat, statistic });
-    });
+    test.each([['Average', 'p23.23', 'p34', 'TM(2%:98%)', 'WM(10%:90%)', 'TS(80%:)', '$statistic']])(
+      'should accept valid values',
+      async (statistic) => {
+        const onChange = jest.fn();
+        props.datasource.getVariables = jest.fn().mockReturnValue(['$statistic']);
+        render(<MetricStatEditor {...props} onChange={onChange} />);
+
+        const statisticElement = await screen.findByLabelText('Statistic');
+        expect(statisticElement).toBeInTheDocument();
+
+        await userEvent.type(statisticElement, statistic);
+        fireEvent.keyDown(statisticElement, { keyCode: 13 });
+        expect(onChange).toHaveBeenCalledWith({ ...props.metricStat, statistic });
+      }
+    );
 
     test.each(['CustomStat', 'p23,23', '$someUnknownValue'])('should not accept invalid values', async (statistic) => {
```

Comment:
Could you add some cases that should not match as well? Currently the regex is matching `tc`, `ts`, `wm`, `tm`, and `pr` even though they are invalid. `pr` doesn't conform to the  form `PR(X:Y)`, `PR(X:)`, or `PR(:Y)`. For the other 4 stats, lowercase letters are only valid when specifying one number as described in https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/Statistics-definitions.html#Statistics-syntax.

Reply:
Added more tests for this and modified the regex expression to handle invalid values